cmd/goplugin-heroku/resource/build.go

```diff
@@ -253,3 +266,25 @@ func uploadSource(filePath, httpMethod, httpUrl string) error {
 
 	return nil
 }
+
+func buildpackCheck(desiredState Build, currentState heroku.Build) bool {
```

Comment:
This function returns `true` when at least one buildpacks exists, when the first build pack either contains or doesn't contain the same name in the first desired versus current build pack, and when the all remaining buildpacks have the same characteristics as the first one. That's an algorithm that is hard to understand . A comment describing the rationale would be helpful.